### PR TITLE
Initial support for I2C on STM32 H7

### PR DESCRIPTION
**Pull request by gmarull**
- Enable I2C driver for H7 series
- Enable I2C on NUCLEO-H743ZI board.

### CODE REVIEW

**Review by FRASTM — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Comment by MaureenHelm** [comment]
Please rebase and fix the merge conflicts

**Review by ydamigos — APPROVED** [review]

File: drivers/pinmux/stm32/pinmux_stm32h7.h [hunk 1]
```diff
@@ -50,19 +54,41 @@
...
 /* Port B */
+#define STM32H7_PINMUX_FUNC_PB6_I2C1_SCL    \
+	(STM32_PINMUX_ALT_FUNC_4 | STM32_OPENDRAIN_PULLUP)
+#define STM32H7_PINMUX_FUNC_PB6_I2C4_SCL    \
+	(STM32_PINMUX_ALT_FUNC_6 | STM32_OPENDRAIN_PULLUP)
 #define STM32H7_PINMUX_FUNC_PB6_USART1_TX   \
 	(STM32_PINMUX_ALT_FUNC_7 | STM32_PUSHPULL_PULLUP)
 #define STM32H7_PINMUX_FUNC_PB6_LPUART1_TX    \
-	(STM32_PINMUX_ALT_FUNC_6 | STM32_OPENDRAIN_PULLUP)
+	(STM32_PINMUX_ALT_FUNC_8 | STM32_OPENDRAIN_PULLUP)
```
Comment (FRASTM) [inline drivers/pinmux/stm32/pinmux_stm32h7.h]: This is an issue, not related to I2C, which could be fixed in another commit

**Review by FRASTM — APPROVED** [review]

File: drivers/pinmux/stm32/pinmux_stm32h7.h (hunk 1 above)
Comment (gmarull) [reply drivers/pinmux/stm32/pinmux_stm32h7.h]: @FRASTM done in 3077a04

**Comment by gmarull** [comment]
@MaureenHelm done

**Pull request merged**